lib/src/common/async_extensions.dart

```diff
@@ -0,0 +1,16 @@
+import 'dart:async';
+
+/// Extensions for [Completer]
+extension CompleterExtensions<T> on Completer<T> {
+  void completeIfNotCompleted([FutureOr<T> value]) {
+    if (!isCompleted) {
+      complete(value);
+    }
+  }
+
+  void completeErrorIfNotCompleted(Object error, [StackTrace stackTrace]) {
```

Comment:
It seems like you might want to use these frequently. In which case, it might be nice to have terser names. Without a lot of thought, I'll suggest `ensureComplete`.

Reply:
I used longer names so that the prefixes match the original names. Ideally, the extension would already be imported and thus users would see this option in autocomplete.